File: API/Entities/Game.cs
namespace API.Entities;

public class Game
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public Platforms Platforms { get; set; }
    public Genres Genres { get; set; }
    public Poster Poster { get; set; }
    public Publication Publication { get; set; }    
    public List<Bookmark> Bookmarks { get; set; } = new();
    public List<Screenshot> Screenshots { get; set; } = new();
    public List<Like> LikedUsers { get; set; } = new();
    public List<Review> Reviews { get; set; } = new();
    public DateTime Release { get; set; } = DateTime.UtcNow;
}

File: client/src/app/models/game.ts
import { Files } from "./files";
import { Genres } from "./genres";
import { Platforms } from "./platforms";
import { Poster } from "./poster";
import { Screenshot } from "./screenshot";

export interface Game {
  id: number;
  title: string;
  description: string;
  platforms: Platforms;
  genres: Genres;
  poster: Poster;
  publisher: string;
  files: Files;
  bookmarks: number[];
  screenshots: Screenshot[];
  likes: number[];
  release: string;
  video: string;
}